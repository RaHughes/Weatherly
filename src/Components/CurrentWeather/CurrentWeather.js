import React from 'react';
import './CurrentWeather.scss'

const CurrentWeather = ({data}) => {
  const date = (data.list[0].dt_txt).split('').slice(0, 10)

  return(
    <article className="current-weather">
        <p> Current City: {data.city.name}</p>
        <p> Current Condition: {data.list[0].weather[0].main}</p>
        <p> Date: {date}</p>
        <p> Temperature: {((data.list[0].main.temp - 273.15) * 9/5 + 32).toFixed(2)} F </p>
        <p> Expected High/Low: {((data.list[0].main.temp_max - 273.15) * 9/5 + 32).toFixed(2)} F / {((data.list[0].main.temp_min - 273.15) * 9/5 + 32).toFixed(2)} F </p>
        <p> Summary: {data.list[0].weather[0].description}</p>
    </article>
  )
}

export default CurrentWeather;