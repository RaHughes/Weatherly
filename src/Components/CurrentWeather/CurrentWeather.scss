.current-weather {
  border: 2px solid black;
}