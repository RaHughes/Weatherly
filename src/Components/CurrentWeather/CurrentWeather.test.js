import React from 'react';
import { shallow } from 'enzyme';
import CurrentWeather from './CurrentWeather';

describe('CurrentWeather', () => {
  it('Should match the Snapshot', () => {
    const mockData = {
      city: { name: 'Denver' },
      list: [
        { main: 'stuff', weather: [ {main: {temp: 255, temp_max: 256, temp_mix: 254} }]}
      ]
    }
    const wrapper = shallow(<CurrentWeather data={mockData} />)
    expect(wrapper).toMatchSnapshot();
  })
})