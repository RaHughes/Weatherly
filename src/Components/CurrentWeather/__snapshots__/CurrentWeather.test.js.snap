// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CurrentWeather Should match the Snapshot 1`] = `ShallowWrapper {}`;
